lib/kvdb/ikvdb.c

```diff
@@ -76,9 +76,12 @@
 
 /* clang-format off */
 
-static_assert((sizeof(uintptr_t) == sizeof(u64)),
+static_assert((sizeof(uintptr_t) == sizeof(uint64_t)),
               "libhse relies on pointers being 64-bits in size");
 
+static_assert((sizeof(atomic_ulong) == sizeof(uint64_t)),
```

Comment:
I feel like it is dead simple to just do the typedef as mentioned in other comments.

Reply:
Simple, but illusory.  My understanding is that the _Atomic keyword can widen the specified type as needed by the implemention to support the various atomic operations.  So creating a typedef with the size in the name doesn't make it so, and I'd rather not go there.   Not to mention that identifiers starting with "atomic_" are reserved, we'd need to call it something like "xatomic_uint64" or some such...  What's not clear to me is the fixed-width types, I suspect that they are only provided if the implementation can provide a type of exactly that width, vs the "least" types that can be whatever size is supported by the implementation that is at least that number of bits.
So, we'd probably want to use the "least" types to ensure things are sufficiently large, but then we to deal with the omf which is using a fixed type, which might then be smaller.
With my patches nothing has changed, we've been using "int" for atomic_t and "long" for atomic64_t, and I'd rather than change that in this series of patches because there are enough changes that I'd prefer it to settle down so that we can verify things are still working properly.  After that, we should probably create a type for seqnos (e.g., xatomic_seqno or some such) and maybe define it as a fixed-width type.  But probably we need to really carefully review the spec regarding C11 atomics, there's a lot of squirrely stuff in there that I don't believe any of use have spent much time with...
What really caught me off guard is that if you want seq_cst semantics (the default) then you can simply apply ++ (for example) to any _Atomic variable and it will perform an atomic increment as if you had called atomic_fetch_add(&myvar, 1).